pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -17,11 +17,175 @@ limitations under the License.
 package injector
 
 import (
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	"k8s.io/client-go/1.5/kubernetes/fake"
+	kapi "k8s.io/kubernetes/pkg/api"
 	"testing"
 )
 
-func TestInject(t *testing.T) {
+func TestInjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred)
+}
+
+func TestInjectTwo(t *testing.T) {
+	binding0, binding1 := getBindings()
+	cred0, cred1 := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding0, cred0)
+	inject(t, injector, binding1, cred1)
+
+	secretsCl := injector.client.Core().Secrets(binding0.Namespace)
+	secret, err := secretsCl.Get(binding0.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred0)
+
+	secretsCl = injector.client.Core().Secrets(binding1.Namespace)
+	secret, err = secretsCl.Get(binding1.Name)
+	if err != nil {
+		t.Fatalf("Error when getting secret: %s", err)
+	}
+	testCredentialsInjected(t, secret.Data, cred1)
+}
+
+func TestUninjectOne(t *testing.T) {
+	binding, _ := getBindings()
+	cred, _ := getCreds()
+
+	injector := fakeK8sBindingInjector()
+	inject(t, injector, binding, cred)
+	injector.Uninject(binding)
+
+	secretsCl := injector.client.Core().Secrets(binding.Namespace)
+	secret, err := secretsCl.Get(binding.Name)
+	if err == nil {
```

Comment:
I suspect my lack of familiarity with the open service broker API and Kubernetes is coming into play here. The reason I didn't make it the way you described was that I thought it was possible for a service to use other keys besides the things in the credential we are trying to uninject.

Is this wrong/not possible? To me it just seemed wrong for us to just delete keys other than the ones in the Credential we are trying to Uninject from our Secret